config/config_test.go

```diff
@@ -2300,3 +2300,42 @@ func TestScrapeConfigDisableCompression(t *testing.T) {
 
 	require.False(t, got.ScrapeConfigs[0].EnableCompression)
 }
+
+func TestScrapeConfigLegacyGlobalImpliesLegacyValidation(t *testing.T) {
```

Comment:
Let's combine all this into a single unit test. Generally we do like

```go
cases := []struct{ ... } {
  {...}, // case 1
  {...}, // case 2
   ...
}

for _, case := range cases {
    t.Run(...)
}
```

You could have test name (to pass in t.Run), expected output (legacy or utf8) and file name in `cases`.